test/vg_statem.erl

```diff
@@ -5,43 +5,135 @@
 -export([command/1, initial_state/0, next_state/3,
          precondition/2, postcondition/3]).
 
--record(state, {topics = #{}}).
+%% sigh
+-export([restart_server/0]).
 
--define(TOPICS, [<<"topic1">>, <<"topic2">>, <<"topic3">>]).
+-record(state,
+        {
+          topics = #{}
+        }).
 
-command(_State) ->
+command(_S = #state{topics = Topics}) ->
+    %% replace with proper oneof when maps are supported
+    {Topic, Info} = one_of(Topics),
+    Index = hwm(Info) + 1,
     frequency(
-      [{1, {call, vg, write, [oneof(?TOPICS), binary()]}},
-       {5, {call, vg, fetch, [oneof(?TOPICS)]}}
+      [%% to tickle close and restart validation bugs
+       {2,   {call, ?MODULE, restart_server, []}},
+       {2,    {call, vg, ensure_topic, [?LET(A, atom(), atom_to_binary(A, utf8))]}},
+       %% write more than is common in typical workloads in order to
+       %% trigger more wraps
+       {40,   {call, vg, write,
+               %%[Topic, message(Topic, Index, binary())]}},
+               [Topic, 0, message(Topic, Index, <<>>)]}},
+       {10,   {call, vg, write,
+               %%[Topic, message(Topic, Index, binary())]}},
+               [Topic, 0, ?LET(I, integer(2, 15),
+                               [message(Topic, Index + N, <<>>)
+                                || N <- lists:seq(0, I)])]}},
+       {40, {call, vg, fetch, [Topic, 0, -1, integer(1, 5)]}},
+       {100, {call, vg, fetch, [Topic, integer(0, Index - 1)]}}
       ]).
 
 %% Initial model value at system start. Should be deterministic.
 initial_state() ->
-    application:ensure_all_started(vonnegut),
-    ok = vg:create_topic(<<"topic1">>),
-    ok = vg:create_topic(<<"topic2">>),
-    ok = vg:create_topic(<<"topic3">>),
-    #state{topics = #{<<"topic1">> => vg:fetch(<<"topic1">>),
-                      <<"topic2">> => vg:fetch(<<"topic2">>),
-                      <<"topic3">> => vg:fetch(<<"topic3">>)}}.
+    application:load(vonnegut),
+    %% decrease these sizes so that the roll machinery will get
+    %% exercised better
+    application:set_env(vonnegut, segment_bytes, 1024),
```

Comment:
You have these set in the config file. Should only do it in one place to not be a potential confusion.